lib/spack/spack/relocate.py

```diff
@@ -415,6 +428,40 @@ def make_binary_placeholder(cur_path_names, allow_root):
         tty.die("Placeholder not implemented for %s" % platform.system())
 
 
+def make_link_placeholder(cur_path_names, cur_dir, old_dir):
+    """
+    Replace old install path with placeholder in absolute links
+    """
+    cur_dir = os.path.realpath(cur_dir)
+    for cur_path in cur_path_names:
+        # realpath is safe, previously ensured absolute
+        placeholder = set_placeholder(spack.store.layout.root)
+        placeholder_prefix = old_dir.replace(spack.store.layout.root,
+                                             placeholder)
+        cur_src = os.path.realpath(cur_path)
+        suffix = cur_src[len(cur_dir):]
```

Comment:
Using indexing to manipulate path strings is fragile; `+`-based concatenation used on the next line is also fragile depending on whether the path contains a `/`.

For extraction of the path, I'd recommend creating a function that splits on `os.sep`; I'll also note that the caller of this function prepends `cur_dir` to each `cur_path` only for this function to dissect it out.

For construction of the new path, I'd also recommend creating a function; `os.path.join` may be helpful, but if the appended path starts with a `/`, note that `os.path.join` will "throw out" the prefix.

Reply:
Using `os.path.relpath` and `os.path.join` allows me to avoid writing my own function, and you're right, it's much stronger code that way.